spec/unit/focusmanager_spec.lua

```diff
@@ -0,0 +1,91 @@
+describe("FocusManager module", function()
+    local FocusManager
+    local layout
+    local Up,Down,Left,Right
+    Up = function(self) self:onFocusMove({0, -1}) end
+    Down = function(self) self:onFocusMove({0, 1}) end
+    Left = function(self) self:onFocusMove({-1, 0}) end
+    Right = function(self) self:onFocusMove({1, 0}) end
+    setup(function()
+        require("commonrequire")
+        FocusManager = require("ui/widget/focusmanager")
+        local Widget = require("ui/widget/textwidget")
+        local w = Widget:new{}
+        layout= {
+                {w,  w,  w},
+                {nil,w,nil},
+                {nil,w,nil},
+                }
+
+    end)
+    it("should go right", function()
+        local focusmanager = FocusManager:new{}
+        focusmanager.layout = layout
+        focusmanager.selected = {y = 1,x = 1}
+        Right(focusmanager)
+        assert.are.same(focusmanager.selected, {y = 1,x = 2})
```

Comment:
It's `expected, object` so you'll have to switch them all around. It doesn't really affect test failure but it'll give confusing error messages otherwise.